### PR TITLE
GUI scale & rainbow support

### PR DESCRIPTION
**Pull request by dominikaaaa**
Turned down GUI scale, added backend support for rainbow GUI outline, increased possible Zoom FOV's

### CODE REVIEW

File: src/main/java/me/zeroeightsix/kami/gui/kami/KamiGUI.java [hunk 1]
```diff
@@ -132,7 +133,6 @@ public void execute(CheckButton component, CheckButtonPoofInfo info) {
             });
             scrollpane.addChild(checkButton);
         }
-
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/KamiGUI.java:135]: Why was this formatting deleted?

File: src/main/java/me/zeroeightsix/kami/gui/kami/KamiGUI.java [hunk 2]
```diff
@@ -52,7 +52,8 @@
  */
 public class KamiGUI extends GUI {
 
-    public static final RootFontRenderer fontRenderer = new RootFontRenderer(1);
+    public static final RootFontRenderer fontRenderer = new RootFontRenderer(0.75F);
+    //public static final RootFontRenderer fontRenderer = new RootFontRenderer(0.65F);
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/KamiGUI.java:56]: There's no need to leave the old line commented out

File: src/main/java/me/zeroeightsix/kami/gui/kami/RootLargeFontRenderer.java [hunk 3]
```diff
@@ -6,8 +6,11 @@
  */
 public class RootLargeFontRenderer extends RootFontRenderer {
 
+    //public RootLargeFontRenderer() {
+    //    super(1.0f);
+    //}
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/RootLargeFontRenderer.java:11]: No need to leave this here

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java [hunk 4]
```diff
@@ -28,7 +31,7 @@
  */
 public class KamiFrameUI<T extends Frame> extends AbstractComponentUI<Frame> {
 
-    ColourHolder frameColour = KamiGUI.primaryColour.setA(100);
+    ColourHolder frameColour = KamiGUI.primaryColour.setA(255);
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java:34]: Why is this 255?

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java [hunk 5]
```diff
@@ -38,18 +41,54 @@
...
+            for (int r=0; r<100; r++) colorsList.add(new Color(r * 255 / 100, 255, 0));
+            for (int g=100; g>0; g--) colorsList.add(new Color(255, g*255/100,0));
+            for (int b=0; b<100; b++) colorsList.add(new Color(255, 0, b * 255 / 100));
+            for (int r=100; r>0; r--) colorsList.add(new Color(r*255/100,0,255));
+            for (int g=0; g<100; g++) colorsList.add(new Color(0,g*255/100,255));
+            for (int b=100; b>0; b--) colorsList.add(new Color(0,255,b*255/100));
+            colorsList.add(new Color(0,255,0));
+            rgbColors = colorsList.toArray(new Color[colorsList.size()]);
+            rgbListGenerated = true;
+        }
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java:71]: Oh my fucking god WHY
Please, just use the RGB code from ActiveModules, it's legit one line.

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java [hunk 6]
```diff
@@ -38,18 +41,54 @@
...
+        if (rgbGuiEnabled)
+        {
+            if (cycleCycleNum == 100-speed || speed > 100)
+            {
+                cycleNum++;
+                cycleCycleNum = 0;
+            }
+            cycleCycleNum++;
+            if (cycleNum > rgbColors.length - 1) cycleNum = 0; //reset once it does a full RGB cycle
+        }
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java:82]: As I said above, the work is already done for you. No need for this

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java [hunk 7]
```diff
@@ -88,20 +127,19 @@ public void renderComponent(Frame component, FontRenderer fontRenderer) {
...
                 glBegin(GL_LINE_LOOP);
                 {
-                    glVertex2d(component.getWidth() - 15, top_y + 2);
-                    glVertex2d(component.getWidth() - 15, bottom_y + 3);
-                    glVertex2d(component.getWidth() - 10, bottom_y + 3);
-                    glVertex2d(component.getWidth() - 10, top_y + 2);
+                    glVertex2d(component.getWidth() - 12, top_y + 2);
+                    glVertex2d(component.getWidth() - 12, bottom_y + 5);
+                    glVertex2d(component.getWidth() - 7, bottom_y + 5);
+                    glVertex2d(component.getWidth() - 7, top_y + 2);
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java:133]: What's the reasoning for these changes?

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java [hunk 8]
```diff
@@ -88,20 +127,19 @@ public void renderComponent(Frame component, FontRenderer fontRenderer) {
...
                 }
-                glEnd();
             } else {
                 glBegin(GL_LINES);
                 {
-                    glVertex2d(component.getWidth() - 15, bottom_y + 4);
-                    glVertex2d(component.getWidth() - 10, bottom_y + 4);
+                    glVertex2d(component.getWidth() - 12, bottom_y + 7.5);
+                    glVertex2d(component.getWidth() - 7, bottom_y + 7.5);
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java:139]: WHere do these numbers come from?

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiGuiColors.java [hunk 9]
```diff
@@ -1,6 +1,8 @@
 package me.zeroeightsix.kami.gui.kami.theme.kami;
 
 import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiGuiColors.java:5]: Why is this here???

File: src/main/java/me/zeroeightsix/kami/module/modules/render/Zoom.kt [hunk 10]
```diff
@@ -2,11 +2,13 @@ package me.zeroeightsix.kami.module.modules.render
 
 import me.zeroeightsix.kami.module.Module
 import me.zeroeightsix.kami.setting.Settings
+import java.math.*
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/render/Zoom.kt:5]: Why is this random import here?

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java (hunk 8 above)
Comment (WolfzWolf) [reply src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java:139]: these numbers are the numbers to fix the minimize button clipping out of the frame with the new font size. I can work on trying to make it dynamic if you want

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java (hunk 7 above)
Comment (WolfzWolf) [reply src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java:133]: fixing the minimize button

File: src/main/java/me/zeroeightsix/kami/gui/kami/KamiGUI.java [hunk 11]
```diff
@@ -193,6 +191,7 @@ public void onMouseDown(MouseButtonEvent event) {
          * Active modules
          */
         Frame frame = new Frame(getTheme(), new Stretcherlayout(1), "Active modules");
+        frame.setName("ActiveModulesFrame");
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/KamiGUI.java:194]: why are you adding a name field when they already have set names???

File: src/main/java/me/zeroeightsix/kami/gui/kami/KamiGUI.java [hunk 12]
```diff
@@ -550,6 +559,7 @@ private static String getEntityName(@Nonnull Entity entity) {
         return entity.getName();
     }
 
+
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/KamiGUI.java:562]: why are you adding extra formatting

File: src/main/java/me/zeroeightsix/kami/gui/kami/KamiGUI.java [hunk 13]
```diff
@@ -516,11 +524,12 @@ public void onTick() {
                 nexty = y;
                 x = 10;
             }
-
             addChild(frame1);
         }
     }
 
+
+
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/KamiGUI.java:532]: why was this added

File: src/main/java/me/zeroeightsix/kami/gui/kami/RootLargeFontRenderer.java [hunk 14]
```diff
@@ -4,10 +4,6 @@
...
-public class RootLargeFontRenderer extends RootFontRenderer {
-
-    public RootLargeFontRenderer() {
-        super(1.2f);
-    }
-
-}
+public class RootLargeFontRenderer extends RootFontRenderer { public RootLargeFontRenderer() {
+        super(0.8f);
+    }}
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/RootLargeFontRenderer.java:9]: formatting please

File: src/main/java/me/zeroeightsix/kami/gui/kami/RootSmallFontRenderer.java [hunk 15]
```diff
@@ -4,10 +4,4 @@
...
  * Created by 086 on 26/06/2017.
  */
-public class RootSmallFontRenderer extends RootFontRenderer {
-
-    public RootSmallFontRenderer() {
-        super(.75f);
-    }
-
-}
+public class RootSmallFontRenderer extends RootFontRenderer { public RootSmallFontRenderer() {super(.75f);}}
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/RootSmallFontRenderer.java:7]: again, fix the formatting :/

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java [hunk 16]
```diff
@@ -14,10 +14,14 @@
...
 import me.zeroeightsix.kami.gui.rgui.util.ContainerHelper;
 import me.zeroeightsix.kami.gui.rgui.util.Docking;
+import me.zeroeightsix.kami.module.modules.client.RGB_Gui;
 import me.zeroeightsix.kami.util.Bind;
 import me.zeroeightsix.kami.util.ColourHolder;
 import me.zeroeightsix.kami.util.Wrapper;
 import org.lwjgl.opengl.GL11;
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java:24]: these imports are completely unnecessary

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java [hunk 17]
```diff
@@ -38,18 +42,17 @@
     boolean centerX = false;
     boolean centerY = false;
     int xLineOffset = 0;
-
     private static final RootFontRenderer ff = new RootLargeFontRenderer();
-
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java:43]: why was this formatting changed

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java [hunk 18]
```diff
@@ -38,18 +42,17 @@
...
     @Override
     public void renderComponent(Frame component, FontRenderer fontRenderer) {
+
         if (component.getOpacity() == 0)
             return;
         glDisable(GL_TEXTURE_2D);
-
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java:49]: ffffformatting

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java [hunk 19]
```diff
@@ -88,20 +91,19 @@ public void renderComponent(Frame component, FontRenderer fontRenderer) {
...
                 glBegin(GL_LINES);
                 {
-                    glVertex2d(component.getWidth() - 15, bottom_y + 4);
-                    glVertex2d(component.getWidth() - 10, bottom_y + 4);
+                    glVertex2d(component.getWidth() - 12, bottom_y + 7.5);
+                    glVertex2d(component.getWidth() - 7, bottom_y + 7.5);
                 }
-                glEnd();
             }
+            glEnd();
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java:106]: why was this changed?

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java [hunk 20]
```diff
@@ -220,6 +222,7 @@ public void onMouseDown(MouseButtonEvent event) {
                         }
                     }
                 }
+
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java:225]: f o r m a t t i n g

File: src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/RootCheckButtonUI.java [hunk 21]
```diff
@@ -30,7 +30,6 @@
 public class RootCheckButtonUI<T extends CheckButton> extends AbstractComponentUI<CheckButton> {
     @Override
     public void renderComponent(CheckButton component, FontRenderer ff) {
-
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/RootCheckButtonUI.java:33]: formatting

File: src/main/java/me/zeroeightsix/kami/gui/rgui/component/AbstractComponent.java [hunk 22]
```diff
@@ -18,6 +18,8 @@
  */
 public abstract class AbstractComponent implements Component {
 
+    String name;
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/rgui/component/AbstractComponent.java:21]: why was this added? components already have names, it's inherited from Component

File: src/main/java/me/zeroeightsix/kami/gui/rgui/component/AbstractComponent.java [hunk 23]
```diff
@@ -136,8 +138,11 @@ public int getHeight() {
         return height;
     }
 
+
+
     boolean workingy = false;
 
+
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/rgui/component/AbstractComponent.java:145]: what the fuck is going on with the formatting

File: src/main/java/me/zeroeightsix/kami/gui/rgui/component/AbstractComponent.java [hunk 24]
```diff
@@ -366,6 +371,12 @@ public Component setMinimumHeight(int height) {
         return this;
     }
 
+    @Override
+    public String getName(){return name;}
+
+    @Override
+    public void setName(String name){this.name = name;}
+
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/rgui/component/AbstractComponent.java:379]: this is completely unnecessary

File: src/main/java/me/zeroeightsix/kami/gui/rgui/component/Component.java [hunk 25]
```diff
@@ -106,4 +106,8 @@
     public int getPriority(); // The higher, the more prioritized.
 
     public void kill();
+
+    public void setName(String name);
+
+    public String getName();
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/rgui/component/Component.java:112]: don't components already have names???

File: src/main/java/me/zeroeightsix/kami/gui/rgui/component/container/AbstractContainer.java [hunk 26]
```diff
@@ -5,11 +5,13 @@
 import me.zeroeightsix.kami.gui.rgui.component.listen.RenderListener;
 import me.zeroeightsix.kami.gui.rgui.poof.use.AdditionPoof;
 import me.zeroeightsix.kami.gui.rgui.render.theme.Theme;
+import me.zeroeightsix.kami.module.modules.client.*;
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/rgui/component/container/AbstractContainer.java:8]: what the frick

File: src/main/java/me/zeroeightsix/kami/gui/rgui/component/container/AbstractContainer.java [hunk 27]
```diff
@@ -30,6 +32,25 @@ public AbstractContainer(Theme theme) {
...
+        {
+            if (c.getName() == null) continue;
+            if (c.getName().equals(name))
+            {
+                c.setVisible(visibility);
+                return;
+            }
+        }
+        throw new NoSuchElementException("The specified element was not found");
+    }
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/gui/rgui/component/container/AbstractContainer.java:52]: could you please format your brackets like so
```java
if (thing) {
```

File: src/main/java/me/zeroeightsix/kami/module/modules/client/Overlays.java [hunk 28]
```diff
@@ -0,0 +1,93 @@
...
+    //lastBoolean added to prevent setting visibility each update
+    private boolean lastRadarDisabled = radarDisabled.getValue();
+    private boolean lastBaritoneDisabled = baritoneDisabled.getValue();
+    private boolean lastActiveModulesDisabled = activeModulesDisabled.getValue();
+    private boolean lastInformationOverlayDisabled = informationOverlayDisabled.getValue();
+    private boolean lastInventoryViewerDisabled = inventoryViewerDisabled.getValue();
+    private boolean lastFriendsListDisabled = friendsListDisabled.getValue();
+    private boolean lastTextRadarDisabled = textRadarDisabled.getValue();
+    private boolean lastEntityListDisabled = entitylistDisabled.getValue();
+    private boolean lastCoordinatesDisabled = coordinatesDisabled.getValue();
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/client/Overlays.java:38]: what the f u c k

File: src/main/java/me/zeroeightsix/kami/module/modules/client/Overlays.java [hunk 29]
```diff
@@ -0,0 +1,93 @@
...
+        lastRadarDisabled = radarDisabled.getValue();
+        lastBaritoneDisabled = baritoneDisabled.getValue();
+        lastActiveModulesDisabled = activeModulesDisabled.getValue();
+        lastInformationOverlayDisabled = informationOverlayDisabled.getValue();
+        lastInventoryViewerDisabled = inventoryViewerDisabled.getValue();
+        lastFriendsListDisabled = friendsListDisabled.getValue();
+        lastTextRadarDisabled = textRadarDisabled.getValue();
+        lastEntityListDisabled = entitylistDisabled.getValue();
+        lastCoordinatesDisabled = coordinatesDisabled.getValue();
+    }
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/client/Overlays.java:66]: you can replace ALL of this with a setting listener

File: src/main/java/me/zeroeightsix/kami/module/modules/client/Overlays.java [hunk 30]
```diff
@@ -0,0 +1,93 @@
...
+        lastRadarDisabled = true;
+        lastBaritoneDisabled = true;
+        lastActiveModulesDisabled = true;
+        lastInformationOverlayDisabled = true;
+        lastInventoryViewerDisabled = true;
+        lastFriendsListDisabled = true;
+        lastTextRadarDisabled = true;
+        lastEntityListDisabled = true;
+        lastCoordinatesDisabled = true;
+    }
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/client/Overlays.java:92]: this shouldn't be changed on disable, instead overlay manager should always be on, like the other client modules

File: src/main/java/me/zeroeightsix/kami/module/modules/client/RGB_Gui.java [hunk 31]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * @author Salt 5/26/2020
+ * RGB Generation by dominikaaaa
+ */
+@Module.Info(
+        name = "RGB Gui",
+        description = "Makes your GUI RGB",
+        category = Module.Category.CLIENT
+)
+public class RGB_Gui extends Module
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/client/RGB_Gui.java:20]: why does your class have an underscore in the name? do like `RainbowGui`

File: src/main/java/me/zeroeightsix/kami/module/modules/client/RGB_Gui.java [hunk 32]
```diff
@@ -0,0 +1,51 @@
...
+
+
+import static me.zeroeightsix.kami.util.ColourConverter.toF;
+
+/**
+ * @author Salt 5/26/2020
+ * RGB Generation by dominikaaaa
+ */
+@Module.Info(
+        name = "RGB Gui",
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/client/RGB_Gui.java:16]: Names are not allowed to have spaces, do `RainbowGui`

File: src/main/java/me/zeroeightsix/kami/module/modules/client/RGB_Gui.java [hunk 33]
```diff
@@ -0,0 +1,51 @@
+package me.zeroeightsix.kami.module.modules.client;
+
+import me.zeroeightsix.kami.module.Module;
+import me.zeroeightsix.kami.setting.Setting;
+import me.zeroeightsix.kami.setting.Settings;
+import java.awt.*;
+
+
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/client/RGB_Gui.java:8]: formatting

File: src/main/java/me/zeroeightsix/kami/module/modules/client/RGB_Gui.java [hunk 34]
```diff
@@ -0,0 +1,51 @@
...
+import static me.zeroeightsix.kami.util.ColourConverter.toF;
+
+/**
+ * @author Salt 5/26/2020
+ * RGB Generation by dominikaaaa
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/client/RGB_Gui.java:13]: 086*

Change that to 
```java
/**
 * @author Salt 5/26/20
 * @see me.zeroeightsix.kami.gui.kami.theme.kami.KamiActiveModulesUI
 */
```

File: src/main/java/me/zeroeightsix/kami/module/modules/client/RGB_Gui.java [hunk 35]
```diff
@@ -0,0 +1,51 @@
...
+    private static int[] rgbList = new int[3];
+
+    public void onEnable()
+    {
+        rgbEnabled = true;
+    }
+    public void onDisable()
+    {
+        rgbEnabled = false;
+    }
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/client/RGB_Gui.java:33]: format this please

File: src/main/java/me/zeroeightsix/kami/module/modules/client/RGB_Gui.java [hunk 36]
```diff
@@ -0,0 +1,51 @@
...
+        rgbList[0] = (rgb >> 16) & 0xFF;
+        rgbList[1] = (rgb >> 8) & 0xFF;
+        rgbList[2] = rgb & 0xFF;
+    }
+    public static boolean guiRgbEnabled() {return rgbEnabled;}
+
+    public static int[] getRgbList()
+    {
+        return rgbList;
+    }
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/client/RGB_Gui.java:49]: why do you have a getter?

File: src/main/java/me/zeroeightsix/kami/module/modules/client/RGB_Gui.java [hunk 37]
```diff
@@ -0,0 +1,51 @@
...
+
+    public void onUpdate()
+    {
+        int newSpeed = rainbowSpeed.getValue();
+        float[] hue = {System.currentTimeMillis() % (360 * newSpeed) / (360f * newSpeed)};
+        int rgb = Color.HSBtoRGB(hue[0], toF(255), toF(255));
+        rgbList[0] = (rgb >> 16) & 0xFF;
+        rgbList[1] = (rgb >> 8) & 0xFF;
+        rgbList[2] = rgb & 0xFF;
+    }
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/client/RGB_Gui.java:43]: don't do this onUpdate, do this when the class is called.

**Comment by dominikaaaa** [comment]
Closed due to inactivity. Make a new pull request once it's ready.
